src/app/project/documents/documents-tab.component.ts

```diff
@@ -315,23 +320,31 @@ export class DocumentsTabComponent implements OnInit, OnDestroy {
     } else {
       this.tableParams.sortBy = '+' + column;
     }
-    this.documentVault = this.sortDocuments(this.documentVault);
+    this.documentVault = this.sortDocuments(this.documentVault, this.tableParams.sortBy, 'sorting');
 		this.tableTemplateUtils.updateUrl(this.tableParams.sortBy, this.tableParams.currentPage, this.tableParams.pageSize, null, this.tableParams.keywords);
 		this.getPaginatedDocs(this.tableParams.currentPage);
   }
 
-	public sortDocuments = (documents: any[]) => {
-		const sortData = this.tableParams.sortBy || '-datePosted';
-		const sortDir = '-' === Array.from(this.tableParams.sortBy)[0] ? -1 : 1;
+	public sortDocuments = (documents: any[], sort: string = '-datePosted', action: string = 'init') => {
```

Comment:
I know this param was `any` to begin with but since you're modifying this function you could quickly set the type to `Document`

```suggestion
	public sortDocuments = (documents: Document[], sort: string = '-datePosted', action: string = 'init') => {
```